scipy/ndimage/tests/test_filters.py

```diff
@@ -2766,3 +2766,40 @@ def test_gh_22333():
     expected = [58, 67, 87, 108, 163, 108, 108, 108, 87]
     actual = ndimage.median_filter(x, size=9, mode='constant')
     assert_array_equal(actual, expected)
+
+
+@pytest.mark.slow
+@pytest.mark.parametrize('length', range(1, 1000, 50))
+@pytest.mark.parametrize('size', range(1, 25))
+@pytest.mark.parametrize('mode', ['reflect', 'constant', 'nearest', 'mirror', 'wrap'])
+def test_gh_22586_median_filter_no_segfault_and_1d_matches_nd(
+    length: int,
+    size: int,
+    mode: str,
+    monkeypatch: pytest.MonkeyPatch,
+):
+    size = min(size, length)
+    arr = np.random.random(length)
+    mf = functools.partial(
+        ndimage.median_filter,
+        size=size,
+        mode=mode,
+    )
+    actual = mf(arr)
+
+    # Monkeypatch _rank_filter_1d.rank_filter() to actually call
+    # _nd_image.rank_filter() so that 1D results from the optimized
+    # implementation can be compared to the general ND implementation
+
+    def _1d_to_nd_rank_filter(i, r, s, *args):
+        from scipy.ndimage import _nd_image
+        return _nd_image.rank_filter(i, r, np.ones(s, dtype=bool), *args)
+
+    with monkeypatch.context():
+        monkeypatch.setattr(
+            'scipy.ndimage._rank_filter_1d.rank_filter',
+            _1d_to_nd_rank_filter,
+        )
```

Comment:
a bit convoluted--I guess there's no easy way to go through a public function directly to check?

Reply:
Yeah there isn't great access to get there without patching or adding another parameter. The closest I got was using an Nx1 array, but that didn't work for `mode='constant'` where it would just output a column of zeros.